esp-hal/src/uart.rs

```diff
@@ -507,6 +520,22 @@ where
         self.tx.write_bytes(data)
     }
 
+    /// Change interpretation of line polarity for inputs/outputs
+    pub fn set_line_invert(&mut self, line: UartLine, invert: bool) {
+        T::register_block().conf0().write(|w| match line {
```

Comment:
I think at least for some chips conf0 needs UART_REG_UPDATE

Reply:
Add sync_regs() to the end to resolve this